### PR TITLE
Add useObservable

### PR DESCRIPTION
**Pull request by ayarushin**
Add `useObservable` hook

## Description
A hook for using `rxjs` observables in react components

## Motivation and Context
Was inspired by how the code looks structured with `rxjs`

## How Has This Been Tested?
- Subscribe and checked single value
- Subscribe and checked multiple values
- Checked the render and the changes in the component.

### CODE REVIEW

**Comment by ayarushin** [comment]
Hope it will be useful 😅

File: package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
   },
   "peerDependencies": {
     "react": ">=16.13.1",
-    "react-dom": ">=16.13.1"
+    "react-dom": ">=16.13.1",
+    "rxjs": "6.6.2"
```
Comment (ayarushin) [inline package.json:45]: Not sure, that it's a good place for it

**Comment by antonioru** [comment]
@ayarushin 
this is actually a great hook and I love the way you wrote it! 
Thank you very much!

File: package.json (hunk 1 above)
Comment (antonioru) [reply package.json:45]: Perhaps this is more a devDependency... 
please fix it :)

File: src/useObservable.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { useEffect } from 'react';
+
+/**
+ * Hook, which helps you combine rxjs flow and setState in your component
+ */
+const useObservable = (observable, setter) => {
+  useEffect(() => {
+    const subscription = observable.subscribe(setter);
+
+    return () => subscription.unsubscribe();
```
Comment (antonioru) [inline src/useObservable.js:10]: Perhaps this could be shortened by:

```
return subscription.unsubscribe;
```

**Review by antonioru — CHANGES_REQUESTED** [review]

**Comment by antonioru** [comment]
@ayarushin you did a great job man, thank you! :)

File: src/useObservable.js (hunk 2 above)
Comment (ayarushin) [reply src/useObservable.js:10]: Yeah, I thought about it, but it works weird with `renderHook` in the test.
It failed with error:
```log
 "after each" hook for "should return a function":
 TypeError: Cannot read property 'closed' of null
```
🤔

Comment (antonioru) [reply src/useObservable.js:10]: All right, never mind then ;)

**Review by antonioru — APPROVED** [review]
Nice and clean! :)

**Pull request merged**